### ISSUE TITLE
TMPDIR

### ISSUE THREAD

**Issue by ciq-mcarifio** [sosreport/sos#2738]
sos report doesn't seem to use the value of TMPDIR to write reports. I understand `--tmp-dir` is a switch and `/etc/sos/sos.conf` will allow for a setting. Still seems like `TMPDIR` should be used. Ty.

**Comment by TurboTurtle** [comment]
Where are you seeing `--tmpdir` not being used? Running against current `main`, it works as expected for me:

```
$ sudo ./bin/sos report --batch -q -o kernel 

sosreport (version 4.2)

Your sosreport has been generated and saved in:
	/var/tmp/sosreport-terra-2021-10-30-shyobvm.tar.xz

$ sudo ./bin/sos report --batch -q -o kernel --tmp-dir=/tmp

sosreport (version 4.2)

Your sosreport has been generated and saved in:
	/tmp/sosreport-terra-2021-10-30-tyuepgz.tar.xz

$ sudo ./bin/sos report --batch -q -o kernel --tmp-dir=/home/jake

sosreport (version 4.2)

Your sosreport has been generated and saved in:
	/home/jake/sosreport-terra-2021-10-30-ebtxocg.tar.xz

```

**Comment by ciq-mcarifio** [comment]
Yup. That works. I was thinking: `TMPDIR=/tmp sudo -E ./bin/sos report --batch -q -o kernel`. Often the environment variable is picked up. But it's not guaranteed.

**Comment by TurboTurtle** [comment]
Hmm. I can see the desire for that, but at least for me I think a cmdline option and a config file setting is sufficient. Though I'm not solidly against it, either. 

@pmoravec @BryanQuigley @bmr-cymru - any thoughts either way?

**Comment by pmoravec** [comment]
`sos` would be nice if it supports `TMPDIR`, as the variable is commonly used as defined in [Single UNIX Specification](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03).

Though I think it is lower prio but it makes sense to support the env.var. If neither cmdline option or config option is specified, we should source `TMPDIR` env.var (or maybe we can use the env.var default, to be overwritten by either cmdline or config option?).
